### PR TITLE
#75941936 Created CRUD functionality

### PR DESCRIPTION
**Pull request by Procech20**
_**What does the PR do?**_
>Database configuration and CRUD functionality setup up
**_Completed task description:_**
-Initiate database into my API
>GET           api/v1/techblogs/blogs/
>GET           api/v1/techblogs/blogs/:id
>POST        api/v1/techblogs/blogs/
>PUT          api/v1/techblogs/blogs/:id
>DELETE     api/v1/techblogs/blogs/:id

>GET         api/v1/techblogs/users/
>GET         api/v1/techblogs/users/:id
>POST       api/v1/techblogs/users/
>PUT         api/v1/techblogs/users/:id
>DELETE    api/v1/techblogs/users/:id

*Am still working on the users routes initialisation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,8 +24,32 @@
   "author": "Procech",
   "license": "MIT",
   "dependencies": {
+    "@babel/polyfill": "^7.12.1",
+    "bcryptjs": "^2.4.3",
+    "body-parser": "^1.19.0",
```
Comment (RedJanvier) [inline package.json:29]: Are you using this package?

File: package.json [hunk 2]
```diff
@@ -17,8 +24,32 @@
...
     "colors": "^1.4.0",
+    "cors": "^2.8.5",
+    "crossenv": "0.0.2-security",
```
Comment (RedJanvier) [inline package.json:32]: I think you are not using this one too!

File: package.json [hunk 3]
```diff
@@ -17,8 +24,32 @@
...
+    "dotenv": "^8.2.0",
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-config-prettier": "^8.0.0",
+    "express": "^4.17.1",
+    "fs": "0.0.1-security",
```
Comment (RedJanvier) [inline package.json:37]: why did you need to install a built in node module?

File: package.json [hunk 4]
```diff
@@ -17,8 +24,32 @@
...
+    "sequelize-cli": "^6.2.0"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.12.17",
+    "@babel/core": "^7.12.17",
+    "@babel/node": "^7.12.17",
+    "@babel/preset-env": "^7.12.17",
+    "eslint": "^5.16.0",
+    "eslint-plugin-import": "^2.22.1",
+    "jshint": "^2.12.0",
```
Comment (RedJanvier) [inline package.json:52]: Please if you are not using a package you need to remove it before we merge.

File: src/App/routes/user.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import User from '../controllers/user';
+
+const router = Router();
+
+router.route('/login').post(User.login);
```
Comment (RedJanvier) [inline src/App/routes/user.js:6]: Move this route to auth Routes please!! same for the controllers.

File: src/Database/models/user.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Model } from 'sequelize';
+import config from '../config/db.config';
+
+
+module.exports = (sequelize, DataTypes) => {
+  class User extends Model {
+
+    static associate(models) {
+      User.hasMany(models.Blog, { foreignKey: 'userId', as: 'Blog'});
```
Comment (RedJanvier) [inline src/Database/models/user.js:9]: you can change this `as` field to a more comprehensive value like `Blogs` because it has many...

File: src/Tests/database.test.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+try {
```
Comment (RedJanvier) [inline src/Tests/database.test.js:1]: This file is not being used too! make sure to remove it or it is being used properly.

File: src/server.js [hunk 8]
```diff
@@ -1,13 +1,51 @@
...
+import auth from './App/routes/auth';
+
+//Load env vars
+dotenv.config();
+
+//Initialising the app with express
 const app = express();
 
-//Body parser
-app.use(express.json());
```
Comment (RedJanvier) [inline src/server.js:7]: why did you change this to use an extra package ?

File: src/server.js [hunk 9]
```diff
@@ -1,13 +1,51 @@
...
+
+//DB Initialisation
+const db = require('./Database/models/server');
 
+//Test DB Connection
+db.sequelize.authenticate()
+    .then( () => console.log('Connection has been established successfully...'.bgGreen.bold))
+    .catch( err => console.error('Unable to connect to the database:'.bgRed.bold, err.message));
 
+db.sequelize.sync();
```
Comment (RedJanvier) [inline src/server.js:46]: Doesn't this delete everything in the database whenever you restart your server? Please find another way.

**Review by RedJanvier — CHANGES_REQUESTED** [review]
Remove all unnecessary packages you aren't using and Also remember to remove the `.gitkeep` which are in non-empty folders please, as the folder is not empty!

File: package.json (hunk 1 above)
Comment (Procech20) [reply package.json:29]: yes, But not yet, I just installed them in advance when I was setting up my app environment

File: package.json (hunk 2 above)
Comment (Procech20) [reply package.json:32]: Oky

File: package.json (hunk 4 above)
Comment (Procech20) [reply package.json:52]: As I told you earlier, these are necessary but am just not using them now but am using them later

File: src/App/routes/user.js (hunk 5 above)
Comment (Procech20) [reply src/App/routes/user.js:6]: Okay

File: src/Database/models/user.js (hunk 6 above)
Comment (Procech20) [reply src/Database/models/user.js:9]: sure

File: src/Tests/database.test.js (hunk 7 above)
Comment (Procech20) [reply src/Tests/database.test.js:1]: Am using it soon I just haven't moved to testing

File: src/Database/models/user.js (hunk 6 above)
Comment (Procech20) [reply src/Database/models/user.js:9]: But I also haven't optimized well my associations since I haven't worked on my user's routes yet

File: src/App/routes/user.js (hunk 5 above)
Comment (Procech20) [reply src/App/routes/user.js:6]: This one is users related route and I haven't handled them well yet!

File: package.json (hunk 3 above)
Comment (Procech20) [reply package.json:37]: Am removing them
